README.rst

```diff
@@ -78,6 +78,28 @@ Want to know if a word you're proposing exists in codespell already? It is possi
     echo "word" | codespell -
     echo "1stword,2ndword" | codespell -
 
+Using a config file
+-------------------
+
+Command line options can also be specific in a config file.
+
+When running ``codespell``, it will check in the current directory for a file
+named ``setup.cfg`` or ``.codespellrc`` (or a file specified via ``--config``),
+containing an entry named ``[tool:codespell]``. Each command line argument can
+also be specified in this file, for example::
+
+    [tool:codespell]
+    skip = *.po,*.ts,./src/3rdParty,./src/Test
+    count =
+    quiet-level = 3
+
+This is equivalent to running::
+
+    codespell --quiet-level 3 --count --skip "*.po,*.ts,./src/3rdParty,./src/Test"
+
+Any options specified in the config file will *override* options from the
+command line.
```

Comment:
As @larsoner mentioned this is the wrong way round and non-standard.

Reply:
There is no standard. Each tool does what it wants. Pytest also has the config override the command line. If you know of an actual standard or PEP please provide it.

Based on the **opinions** voiced here it can be changed the other way around.

Reply:
> Pytest also has the config override the command line.

This does not appear to be true, at least for the `addopts` option in `setup.cfg`. We have in `setup.cfg` `addopts = ... -rs`, so get this with no command-line options:

```
larsoner@bunk:~/python/codespell$ pytest codespell_lib/tests/test_dictionary.py 
=============================================================================== test session starts ================================================================================
platform linux -- Python 3.8.2, pytest-5.4.3, py-1.8.1, pluggy-0.13.0
PyQt5 5.15.0 -- Qt runtime 5.15.0 -- Qt compiled 5.15.0
rootdir: /home/larsoner/python/codespell, inifile: setup.cfg
plugins: notebook-0.6.0, qt-3.3.0, dependency-0.5.1, cov-2.8.1, mock-3.2.0, timeout-1.3.3
collected 43 items                                                                                                                                                                 

codespell_lib/tests/test_dictionary.py ...........................................                                                                                           [100%]
```

But when passing `-rA` we get (A)ll output:

<details>

```
================================================================================ 43 passed in 2.70s ================================================================================
platform linux -- Python 3.8.2, pytest-5.4.3, py-1.8.1, pluggy-0.13.0
PyQt5 5.15.0 -- Qt runtime 5.15.0 -- Qt compiled 5.15.0
rootdir: /home/larsoner/python/codespell, inifile: setup.cfg
plugins: notebook-0.6.0, qt-3.3.0, dependency-0.5.1, cov-2.8.1, mock-3.2.0, timeout-1.3.3
collected 43 items                                                                                                                                                                 

codespell_lib/tests/test_dictionary.py ...........................................                                                                                           [100%]

====================================================================================== PASSES ======================================================================================


============================================================================= short test summary info ==============================================================================
PASSED codespell_lib/tests/test_dictionary.py::test_dictionaries_exist
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_formatting[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary.txt-in_aspell0]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_formatting[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_rare.txt-in_aspell1]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_formatting[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_informal.txt-in_aspell2]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_formatting[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_usage.txt-in_aspell3]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_formatting[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_code.txt-in_aspell4]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_formatting[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_names.txt-in_aspell5]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_formatting[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_en-GB_to_en-US.txt-in_aspell6]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a a-bar-has whitespace]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a,a-bar-has a comma]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a--non-empty]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a- bar-start with whitespace]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a-,bar-starts with a comma]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a-bar,bat-.*not.*followed by a space]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a-bar -trailing space]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a-b ,ar-contains a whitespace.*followed by a comma]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a-bar,-single entry.*comma]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a-bar, bat-must end with trailing ","]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a-a, bar,-corrects to itself amongst others]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a-a-corrects to itself]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking[a-bar, bar,-unique]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[abcdef-uvwxyz, bar,-True-None-should be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[a-uvwxyz, bar,-False-None-should not be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[a-abcdef, uvwxyz,-None-True-should be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[abcdef-uvwxyz, bar,-True-True-should be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[abcdef-uvwxyz, bar,-False-True-should be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[a-bar, back,-None-False-should not be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[abcdef-ghijkl, uvwxyz,-True-False-should be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[abcdef-uvwxyz, bar,-False-False-should not be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[a-abcdef uvwxyz-None-True-should be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[a-bar back-None-False-should not be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[a-bar back, abcdef uvwxyz, bar,-None-True-should be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[a-abcdef uvwxyz, bar back, ghijkl,-None-False-should not be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[a-bar abcdef-None-True-should be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_error_checking_in_aspell[a-abcdef back-None-False-should not be in aspell]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_looping[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary.txt-in_aspell0]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_looping[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_rare.txt-in_aspell1]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_looping[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_informal.txt-in_aspell2]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_looping[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_usage.txt-in_aspell3]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_looping[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_code.txt-in_aspell4]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_looping[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_names.txt-in_aspell5]
PASSED codespell_lib/tests/test_dictionary.py::test_dictionary_looping[/home/larsoner/python/codespell/codespell_lib/tests/../data/dictionary_en-GB_to_en-US.txt-in_aspell6]
PASSED codespell_lib/tests/test_dictionary.py::test_ran_all
================================================================================ 43 passed in 2.71s ================================================================================

```

</details>

i.e., the command-line overrides `setup.cfg`. I'm pretty sure I've done it in the past with other pytest options, too, and other tools like flake8 and pydocstyle, etc. I've also done it with `ssh` and other Linux command-line tools `.ssh/config`, etc. -- so I consider it to be fairly standard behavior. But I could be wrong!